src/compiler/scala/reflect/macros/compiler/Validators.scala

```diff
@@ -79,7 +79,7 @@ trait Validators {
         val boundsOk = typer.silent(_.infer.checkBounds(macroDdef, NoPrefix, NoSymbol, atparams, atargs, ""))
         boundsOk match {
           case SilentResultValue(true) => // do nothing, success
-          case SilentResultValue(false) | SilentTypeError(_) => MacroImplTargMismatchError(atargs, atparams)
+          case SilentResultValue(false) | _: SilentTypeError => MacroImplTargMismatchError(atargs, atparams)
```

Comment:
interesting style preference (compared to other case, where there is a binding). I see that it's actually an extractor not a constructor pattern. Now I wonder why it doesn't warn that the extractor takes an error not a result. Is it smart enough to know that both types are covered by the case?

Reply:
Yeah.  And I guess it considers the extractor taking a subtype rather than the scrutinee type as the first reason to makes it refutable given any result value.

Reply:
Btw, seems to me like this could do with `SilentTypeError` being irrefutable, by guaranteeing by construction that it wraps at least 1 error.  Not necessarily, but maybe it's long overdue we have a compiler-local NEL?